### DISCUSSION TITLE
Upload to Minio works, but download URL points to s3.amazonaws.com

### DISCUSSION THREAD

**Discussion by jonasjancarik** [outline/outline#5616] [Self Hosting] [answered]
I set up Outline on a local server using Docker. Uploading a file to Minio works, but when I try to download the file, Outline redirects me to a URL starting with `https://s3.amazonaws.com/outline/uploads/...` instead of the local Minio URL.

What I have in the .env file:

```
AWS_ACCESS_KEY_ID=minioadmin
AWS_SECRET_ACCESS_KEY=minioadmin
AWS_S3_UPLOAD_BUCKET_URL=http://localhost:4586/outline
AWS_S3_UPLOAD_BUCKET_NAME=outline
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_S3_FORCE_PATH_STYLE=true
```

I am using the default Minio credentials for testing.

What am I doing wrong here?

**Comment by jonasjancarik** [comment, marked as answer]
I think the issue was including the bucket name in the bucket URL. Like this it works:
```
AWS_S3_UPLOAD_BUCKET_URL=http://localhost:4586
AWS_S3_UPLOAD_BUCKET_NAME=outline
```

**Reply by joeyame** [reply]
When I use this same config it breaks uploading because it isn't appending the bucket name to the bucket url

But, if I do include the bucket name in the url, it uses two repeats of the bucket name in the download URL and that breaks it.

**Reply by jonasjancarik** [reply]
How did you set up the minio bucket in the first place? I just checked and this config is working for me, both for uploads and downloads.

I think I used `mc alias set outline http://localhost:4586 <user> <password>` and then `mc mb outline/outline` (so that's creating a bucket called `outline` on a server with the alias `outline`).

**Reply by joeyame** [reply]
Okay, I deleted my minio data and restarted, following your config options and this blog: https://blog.yarsalabs.com/self-hosting-outline-wiki-on-cloudron/ and now it works! Thank you for sharing!

**Reply by jonasjancarik** [reply]
No problem, good to hear you got it working. I can also recommend this repo for setting up self-hosted Outline https://github.com/vicalloy/outline-docker-compose I like that it sets up an authentication server for you so you don't have to rely on a third-party solution.
